promotion/src/main/java/com/yas/promotion/repository/PromotionRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.yas.promotion.repository;
+
+import com.yas.promotion.model.Promotion;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface PromotionRepository extends JpaRepository<Promotion, Long> {
+    Optional<Promotion> findBySlugAndIsActiveTrue(String slug);
+
+//    @Query("select p from Promotion p where p.name = :name and (:id is null or p.id != :id)")
+//    Promotion findExistedName(String name, Long id);
```

Comment:
remove unuse code or if you want to find exist there is a better way https://stackoverflow.com/questions/30392129/spring-data-jpa-and-exists-query